restapi/src/main/java/io/stargate/web/docsapi/exception/ErrorCodeRuntimeException.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.stargate.web.docsapi.exception;
+
+import java.util.Objects;
+import javax.ws.rs.core.Response;
+
+/** {@link RuntimeException} that carries an {@link ErrorCode}. */
+public class ErrorCodeRuntimeException extends RuntimeException {
+
+  /** Dedicated {@link ErrorCode}. */
+  private final ErrorCode errorCode;
+
+  /** ErrorCode only constructor. */
+  public ErrorCodeRuntimeException(ErrorCode errorCode) {
+    this.errorCode = errorCode;
+  }
+
+  /** @see Exception#Exception(String) */
+  public ErrorCodeRuntimeException(ErrorCode errorCode, String message) {
+    super(message);
+    this.errorCode = errorCode;
+  }
+
+  /** @see Exception#Exception(String, Throwable) */
+  public ErrorCodeRuntimeException(ErrorCode errorCode, String message, Throwable cause) {
+    super(message, cause);
+    this.errorCode = errorCode;
+  }
+
+  /** Pipe to {@link ErrorCode#toResponse()} with the exception message. */
+  public Response getResponse() {
+    return errorCode.toResponse(getLocalizedMessage());
```

Comment:
I'm not sure about `getLocalizedMessage` in this context. It give the impression of producing a localized message for the end user, but will use the server's default locale, which is probably not user-specific.

Reply:
Fine, will adapt..